Source/SwiftLintFramework/Extensions/SyntaxMap+SwiftLint.swift

```diff
@@ -11,11 +11,51 @@ extension SyntaxMap {
                 .intersects(byteRange)
         }
 
-        guard let startIndex = tokens.firstIndex(where: intersect) else {
+        guard let startIndex = firstIntersectingTokenIndex(inByteRange: byteRange) else {
             return []
         }
-        let tokensBeginningIntersect = tokens.lazy.suffix(from: startIndex)
-        return Array(tokensBeginningIntersect.filter(intersect))
+
+        let intersectingTokens = tokens
+            .lazy
+            .suffix(from: startIndex)
+            .prefix(while: intersect)
```

Comment:
one thing to keep in mind is that we changed this to fix a false positive: https://github.com/realm/SwiftLint/pull/2229

Reply:
It seems that this one changed in Xcode11 while still true fro Xcode 10.

Can we know what xcode is used? I think it's SourceKit (version) which matters.

I still don't want to filter it, but can be that adding additional N tokens should help? If this happens for string interpolation, can we assume that there are no more than N string interpolations in the String?

Reply:
Or, just check on Xcode 10 which Tokens are returned within that order (non-ascending)

Reply:
> It seems that this one changed in Xcode11 while still true fro Xcode 10.
> 
> Can we know what xcode is used? I think it's SourceKit (version) which matters.
> 
You can check `SwiftVersion.current`. The compiler version is what matters

> I still don't want to filter it, but can be that adding additional N tokens should help? If this happens for string interpolation, can we assume that there are no more than N string interpolations in the String?

I'd vote against it. Correctness is more important than speed in this case.

Reply:
And in fact, this change didn't seem to cause too severe regressions according to my past self: https://github.com/realm/SwiftLint/pull/2229#issuecomment-394137732

Reply:
It is hard to know what time spent on the shole linting process. Most of the time will be spent on File I/O operations. For this only I checked this solution by measuring function running time on OSS products.
If we look at the winning times, those are within 1-3% or less of total project liting time.
In the measurings above `map` was used instead of `parallelMap` in order to prevent race condition in compilation time. This technically means that  the total winning time on the project should be divided by the number of processes.
By that - we definitely  can say that there's a win in time, but it is not that big as we want it to be

In short. the win is not that big. There should be few more PRs like this one in order to see the results )

Reply:
Please note that the timings above are in total CPU time. So for the full project run this time will be divided between cores